internal/controller/nginx/config/split_clients.go

```diff
@@ -23,7 +24,75 @@ func executeSplitClients(conf dataplane.Configuration) []executeResult {
 	return []executeResult{result}
 }
 
-func createSplitClients(backendGroups []dataplane.BackendGroup) []http.SplitClient {
+func collectAllSplitClients(conf dataplane.Configuration) []http.SplitClient {
+	var splitClients []http.SplitClient
+
+	splitClients = append(splitClients, createBackendGroupSplitClients(conf.BackendGroups)...)
+	splitClients = append(splitClients, createRequestMirrorSplitClients(conf.HTTPServers)...)
+	splitClients = append(splitClients, createRequestMirrorSplitClients(conf.SSLServers)...)
+	splitClients = removeDuplicateSplitClients(splitClients)
+
+	return splitClients
+}
+
+func createRequestMirrorSplitClients(servers []dataplane.VirtualServer) []http.SplitClient {
+	var splitClients []http.SplitClient
+
+	for _, server := range servers {
+		mirrorPathToPercentage := extractMirrorTargetsWithPercentages(server.PathRules)
+
+		for _, pathRule := range server.PathRules {
+			mirrorPercentage, exists := mirrorPathToPercentage[pathRule.Path]
+
+			if mirrorPercentage != 100 && exists {
+				splitClient := http.SplitClient{
+					// this has to be something unique and able to be accessed from the server side
```

Comment:
```suggestion
					// this has to be something unique and able to be accessed from the server block
```
I'm assuming this is what you mean, and I think this is clearer.